forge-game/src/main/java/forge/game/card/CardFactoryUtil.java

```diff
@@ -1518,7 +1523,10 @@ public static void addTriggerAbility(final KeywordInterface inst, final Card car
             final String manacost = k[1];
             final String abStrReveal = "DB$ Reveal | Defined$ You | RevealDefined$ Self"
                     + " | MiracleCost$ " + manacost;
-            final String abStrPlay = "DB$ Play | Defined$ Self | Optional$ True | PlayCost$ " + manacost;
+            String abStrPlay = "DB$ Play | Defined$ Self | Optional$ True | PlayCost$ " + manacost;
```

Comment:
needs ValidSA$ so you can't play enchantment land 🐛

Reply:
the branch for `tgtSA.isLandAbility()` happens before that though

Reply:
hm imo the isLandAbility Check there should check if `"ManaCost".equals(sa.getParam("PlayCost"))`

Reply:
i added a check to isLandAbility part

i wanted to move the parts even more up, like when spells and cards are checked, but i failed with the java syntax a bit

Reply:
Well I don't see how that's better since for something like _Strength of the Harvest_ you shouldn't even be allowed to choose land backside to confuse things
but it's your PR ;)

Reply:
i'm trying to move the check up

right now, i added the check to the whole SpellAbility List